src/utils/decodeFunctionSignature.ts

```diff
@@ -0,0 +1,29 @@
+export interface DecodedFunctionSignature {
+  function: string | undefined;
+  contract: string | undefined;
+}
+
+export const decodeFunctionSignature = (signature: string): DecodedFunctionSignature => {
```

Comment:
This is a very specific function so we should rename it for it's specific use case. Also, it should probaby throw an error in the 'default' case instead of returning null. No?

Reply:
We might want to move it out of utils and into the Trading contract, no?